packages/client/src/views/Home/RecordAudit.tsx

```diff
@@ -0,0 +1,608 @@
+  }
+`
+
+const KeyContainer = styled.div`
+  width: 190px;
+  color: ${({ theme }) => theme.colors.grey};
+  ${({ theme }) => theme.fonts.bodyBoldStyle}
+`
+
+const ValueContainer = styled.div<{ value: undefined | string }>`
+  width: 325px;
+  color: ${({ theme, value }) =>
+    value ? theme.colors.grey : theme.colors.grey600};
+  ${({ theme }) => theme.fonts.captionBigger};
+`
+
+const GreyedInfo = styled.div`
+  height: 26px;
+  background-color: ${({ theme }) => theme.colors.greyInfo};
+  max-width: 330px;
+`
+
+interface IStateProps {
+  workqueue: IWorkqueue
+  resources: IOfflineData
+  savedApplications: IApplication[]
+}
+
+interface IDispatchProps {
+  goToApplicationDetails: typeof goToApplicationDetails
+}
+
+type IFullProps = IDispatchProps &
+  IStateProps &
+  IDispatchProps &
+  IntlShapeProps & { theme: ITheme } & RouteComponentProps<{
+    applicationId: string
+  }>
+
+interface ILabel {
+  [key: string]: string | undefined
+}
+
+interface IApplicationData {
+  id: string
+  name?: string
+  status?: string
+  trackingId?: string
+  type?: string
+  dateOfBirth?: string
+  dateOfDeath?: string
+  placeOfBirth?: string
+  placeOfDeath?: string
+  informant?: string
+  informantContact?: string
+  brnDrn?: string
+}
+
+interface IGQLApplication {
+  id: string
+  child?: { name: GQLHumanName[] }
+  deceased?: { name: GQLHumanName[] }
+  registration?: {
+    trackingId: string
+    type: string
+    status: { type: string }[]
+  }
+}
+
+const STATUSTOCOLOR: { [key: string]: string } = {
+  DRAFT: 'violet',
+  DECLARED: 'orange',
+  REJECTED: 'red',
+  VALIDATED: 'orange',
+  REGISTERED: 'green',
+  CERTIFIED: 'green',
+  WAITING_VALIDATION: 'teal'
+}
+
+const KEY_LABEL: ILabel = {
+  status: 'Status',
+  type: 'Event',
+  trackingId: 'Tracking ID',
+  dateOfBirth: 'Date of birth',
+  dateOfDeath: 'Date of death',
+  placeOfBirth: 'Place of birth',
+  placeOfDeath: 'Place of death',
+  informant: 'Informant',
+  brn: 'BRN',
+  drn: 'DRN'
+}
+
+const NO_DATA_LABEL: ILabel = {
+  status: 'No status',
+  type: 'No event',
+  trackingId: 'No tracking id',
+  dateOfBirth: 'No date of birth',
+  dateOfDeath: 'No date of death',
+  placeOfBirth: 'No place of birth',
+  placeOfDeath: 'No place of death',
+  informant: 'No informant'
+}
+
+const getCaptitalizedword = (word: string | undefined): string => {
+  word = word && word.toUpperCase()[0] + word.toLowerCase().slice(1)
+  return word || ''
+}
+
+const isBirthApplication = (
+  application: GQLEventSearchSet | null
+): application is GQLBirthEventSearchSet => {
+  return (application && application.type === 'Birth') || false
+}
+
+const isDeathApplication = (
+  application: GQLEventSearchSet | null
+): application is GQLDeathEventSearchSet => {
+  return (application && application.type == 'Death') || false
+}
+
+const getDraftApplicationName = (application: IApplication): string => {
+  let name = ''
+  const applicationName = application.data.child || application.data.deceased
```

Comment:
Does the application have `application.type` property as this point you could use to select where to read the name from?

Reply:
You could use the `isBirthApplication` you've made